Auto doc

KarateCowboy opened the pull request.

dickeyxxx commented on circle.yml:
```diff
@@ -29,7 +29,14 @@ stages:
           echo $HEROKU_DEB_KEY | base64 -d | gpg --import
+          echo "machine api.heroku.com" >> ~/.netrc
+          echo "    password $HEROKU_API_KEY" >> ~/.netrc
+          echo "    jeff+devcenter@heroku.com" >> ~/.netrc
```
can you take this out? it shouldn't matter what the login string is

dickeyxxx commented on circle.yml:
```diff
@@ -29,7 +29,14 @@ stages:
         command: |
           set -e
           echo $HEROKU_DEB_KEY | base64 -d | gpg --import
+          echo "machine api.heroku.com" >> ~/.netrc
```
use a heredoc instead to make this cleaner

KarateCowboy commented:
Bump bump. Need another review

dickeyxxx commented on circle.yml:
```diff
@@ -30,12 +30,20 @@ stages:
           echo $HEROKU_DEB_KEY | base64 -d | gpg --import
+          cat << EOF > ~/.netrc
+          machine api.heroku.com
+            login jeff+devcenter@heroku.com 
```
can you take this email out? it should work without it

dickeyxxx replied:
you can just make it junk

dickeyxxx replied:
this is the third time I've asked for this...

KarateCowboy replied:
You may recall I actually did a screencast of that not working on my local machine and posted it in slack. Suffice to say I had some trouble with it.  It's working CI now. I don't know what changed, but it wasn't working before.

dickeyxxx commented on package.json:
```diff
@@ -1,61 +1,64 @@
-    "heroku-orgs": "1.6.4",
-    "heroku-pg": "2.2.5",
-    "heroku-pipelines": "2.1.3",
+	"name": "heroku-cli",
```
can you change this indentation back?

dickeyxxx commented on package.json:
```diff
@@ -1,61 +1,64 @@
     "semver": "5.3.0"
   },
+  "devDependencies": {
+    "heroku-plugin-readme-generator": "git+ssh://git@github.com/heroku/heroku-cli-dev-center-generator.git#flags-as-tables"
```
we should use the master branch

KarateCowboy commented:
OK.

I removed the tabs for that spacing, changed the dependency to point to the master branch, and changed the circle.yml file to dump some junk instead of a valid user name.

KarateCowboy commented:
@dickeyxxx ready for review again

dickeyxxx commented on package.json:
```diff
@@ -56,6 +56,9 @@
     "semver": "5.3.0"
   },
+  "devDependencies": {
+    "heroku-plugin-readme-generator": "git+ssh://git@github.com/heroku/heroku-cli-dev-center-generator.git"
```
CI is failing on appveyor because it doesn't seem to work with a git reference. I published this to npm though, so change this reference to use the package and it should be resolved

dickeyxxx commented on .circleci/config.yml:
```diff
@@ -15,12 +15,31 @@ jobs:
         key: cli
       - type: shell
         command: |
+          set -e
```
this should go as a separate workflow job to run it in parallel. Also, I think this could be cleaned up with the new git tag stuff: https://circleci.com/docs/2.0/workflows/#git-tag-job-execution

dickeyxxx commented on .circleci/config.yml:
```diff
@@ -43,6 +43,33 @@ jobs:
+      - type: shell
+        command: |
+          set -e
+          yarn config set registry https://cli-npm.heroku.com
```
this shouldn't be needed since the .yarnrc file already sets this

dickeyxxx commented on .circleci/config.yml:
```diff
@@ -43,6 +43,33 @@ jobs:
+          set -e
+          yarn config set registry https://cli-npm.heroku.com
+          yarn install
+          yarn check
```
no need to run this

dickeyxxx commented on .circleci/config.yml:
```diff
@@ -43,6 +43,33 @@ jobs:
+  generate_dev_center_docs:
+    <<: *defaults
+    filters:
+      tags:
```
should be able to use this https://github.com/heroku/cli/blob/master/.circleci/config.yml#L11
